engine/src/main/java/org/terasology/rendering/dag/nodes/RefractiveReflectiveBlocksNode.java

```diff
@@ -267,4 +286,37 @@ public void update() {
 
         readOnlyGBufferFbo.attachDepthBufferTo(refractiveReflectiveFbo);
     }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent event) {
```

Comment:
This method is an _almost_ exact duplicate of other propertyChange methods. Perhaps, rather than inheriting the method from an ancestor class (and here overriding it because of the animated water aspect) like I suggest in a previous comment, we could have a couple of more specialized methods in an ancestor class, so that you could write:

```java
if (event.getPropertyName().equals(RenderingConfig.NORMAL_MAPPING)) {
    handleNormalMappingSettingChange();
} else if (event.getPropertyName().equals(RenderingConfig.PARALLAX_MAPPING)) {
    handleParallaxMappingSettingChange();
} else if (event.getPropertyName().equals(RenderingConfig.ANIMATE_WATER)) {
   animateWaterIsEnabled = renderingConfig.isAnimateWater();
} // else: no other case is possible. See subscribe operations in initialise().
```